### PR TITLE
Fix dm scroll position when revisiting a thread

### LINKED ISSUE #4920: DM thread does not scroll back to bottom if you leave and then return
@mxstbr this is generally just a problem with how we approached scroll behavior in the entire app: if a thing loads for the first time, we handle the scroll properly. But then if the user leaves and comes back, the effect or method doesn't fire because it doesn't think anything is new (reads from cache) and so the scroll handler never fires...this applies to chat tab, thread messages, dm messages

### PR DESCRIPTION
**Pull request by brianlovin**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hyperion (frontend)

Closes #4920

### CODE REVIEW

**Comment by mxstbr** [comment]
Won't this break if doing anything in the DM thread that causes a re-render?

**Comment by brianlovin** [comment]
You're right. I've now accounted for a change in loading state from loading => not loading. This appears to be working locally

**Review by mxstbr — DISMISSED** [review]
Let's try it, although I fear that this may misfire if paginating and/or refetching in the background

**Comment by brianlovin** [comment]
Good call, let me think of a different approach.

File: src/views/directMessages/components/messages.js [hunk 1]
```diff
@@ -36,6 +36,14 @@ class MessagesWithData extends React.Component<Props, State> {
   subscription: ?Function;
 
   componentDidMount() {
+    const thread = this.props.data.directMessagethread;
```
Comment (mxstbr) [inline src/views/directMessages/components/messages.js]:
```suggestion
    const thread = this.props.data.directMessageThread;
```

**Review by mxstbr — DISMISSED** [review]

**Pull request merged**